python/spinner_widget/spinner_widget.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+A simple widget that draws an animated spinner that spins whilst visible.
+"""
+
+import sgtk
+from sgtk.platform.qt import QtCore, QtGui
+
+import time
+import math
+
+class SpinnerWidget(QtGui.QWidget):
+    """
+    Spinner widget class
+    """
+    def __init__(self, parent=None):
+        """
+        Construction
+
+        :param parent:    The parent widget
+        """
+        QtGui.QWidget.__init__(self, parent)
+
+        # public properties:
+        self.fps = 20
+        self.border = 2
+        self.line_width = 2
+        self.arc_length = 300
+        self.seconds_per_spin = 3
+
+        # timer used to force a repaint:
+        self._timer = QtCore.QTimer(self)
```

Comment:
To be clear, I agree with the current implementation, let's not overcomplicate things at first. I was simply curious.